react/features/chat/functions.js

```diff
@@ -1,5 +1,49 @@
 // @flow
 
+import escapeRegexp from 'escape-string-regexp';
+import aliases from 'react-emoji-render/data/aliases';
+import emojiAsciiAliases from 'react-emoji-render/data/asciiAliases';
+
+/**
+ * An ASCII emoticon regexp array to find and replace old-style ASCII
+ * emoticons (such as :O) to new Unicode representation, so then devices
+ * and browsers that support them can render these natively without
+ * a 3rd party component.
+ *
+ * NOTE: this is currently only used on mobile, but it can be used
+ * on web too once we drop support for browsers that don't support
+ * unicode emoji rendering.
+ */
+const EMOTICON_REGEXP_ARRAY = [];
+
+(function() {
+    for (const [ key, value ] of Object.entries(emojiAsciiAliases)) {
+        // $FlowExpectedError
+        const escapedValues = value.map(v => escapeRegexp(v));
+        const regexp = `(${escapedValues.join('|')})`;
+        const replaceValue = aliases[key];
+
+        EMOTICON_REGEXP_ARRAY.push([ regexp, replaceValue ]);
+    }
+})();
+
+/**
+ * Replaces ascii emoticons with unicode emojis to let the emojis be rendered
+ * by the platform native renderer.
+ *
+ * @param {string} message - The message to parse and replace.
+ * @returns {string}
+ */
+export function replaceAsciiEmoticon(message: string) {
```

Comment:
lets call this replaceEmoticon since it can also work with the non ASCII (even though they are ascii, just not old style)